### PR TITLE
GCLOUD2-11796 manage servergroup

### PR DESCRIPTION
**Pull request by alexk53**

### CODE REVIEW

File: docs/resources/servergroup.md [hunk 1]
```diff
@@ -12,29 +11,139 @@ Represent server group resource
...
+  project_id = data.gcore_project.project.id
+  region_id  = data.gcore_region.region.id
+}
+
+resource "gcore_instancev2" "instance2" {
+  flavor_id     = "g1-standard-2-4"
+  name          = "my-instance2"
+
+  volume {
+    volume_id  = gcore_volume.boot_volume.id
```
Comment (IsabellaMerkulova) [inline docs/resources/servergroup.md:92]: boot_volume2

File: docs/resources/servergroup.md [hunk 2]
```diff
@@ -12,29 +11,139 @@ Represent server group resource
...
+    boot_index = 0
+  }
+
+  interface {
+    type = "external"
+    name = "my-external-interface"
```
Comment (IsabellaMerkulova) [inline docs/resources/servergroup.md:98]: should it be "my-external-interface2"?

File: docs/resources/servergroup.md [hunk 3]
```diff
@@ -12,29 +11,139 @@ Represent server group resource
...
 - `name` (String) Displayed server group name
-- `policy` (String) Server group policy. Available value is 'affinity', 'anti-affinity'
+- `policy` (String) Server group policy. Available value is 'affinity' 'anti-affinity' 'soft-anti-affinity'. 
+
+**Affinity rule**:
+
+Boost performance by grouping related instances together. They will be hosted on the same physical server for
+faster interaction, enhancing communication and resource sharing.
+
+**Anti-affinity soft rule**:
```
Comment (IsabellaMerkulova) [inline docs/resources/servergroup.md:134]: Anti-affinity soft rule -> Soft anti-affinity rule?

File: docs/resources/servergroup.md [hunk 4]
```diff
@@ -12,29 +11,139 @@ Represent server group resource
...
+
+Balance separation and flexibility, occasionally placing instances on different physical servers for fault tolerance,
+or on the same server physical server for faster interaction and resource optimization.
```
Comment (IsabellaMerkulova) [inline docs/resources/servergroup.md:137]: This description is not accurate
This one from Openstack doc reflects the reality:
Attempts to restrict instances belonging to the server group to the same host. Where it is not possible to schedule all instances on one host, they will be scheduled together on as few hosts as possible.

File: docs/resources/servergroup.md (hunk 2 above)
Comment (IsabellaMerkulova) [reply docs/resources/servergroup.md:98]: or should it be unique only inside one instance?

File: docs/resources/servergroup.md [hunk 5]
```diff
@@ -12,29 +11,139 @@ Represent server group resource
...
+
+**Anti-affinity**: 
+
+Strictly isolate instances to different physical servers, minimizing simultaneous failures for critical applications.
+Choose this for enhanced fault tolerance.
```
Comment (IsabellaMerkulova) [inline docs/resources/servergroup.md:142]: anti-affinity: Restricts instances belonging to the server group to separate hosts.
you mixed up the descriptions of Anti-affinity and affinity

File: examples/resources/gcore_servergroup/with-two-instances.tf [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  project_id = data.gcore_project.project.id
+  region_id  = data.gcore_region.region.id
+}
+
+resource "gcore_instancev2" "instance2" {
+  flavor_id     = "g1-standard-2-4"
+  name          = "my-instance2"
+
+  volume {
+    volume_id  = gcore_volume.boot_volume.id
```
Comment (IsabellaMerkulova) [inline examples/resources/gcore_servergroup/with-two-instances.tf:48]: boot_volume2

File: gcore/resource_gcore_servergroup.go [hunk 7]
```diff
@@ -80,20 +88,37 @@ func resourceServerGroup() *schema.Resource {
 				ForceNew:    true,
 			},
 			"policy": {
-				Type:        schema.TypeString,
-				Description: "Server group policy. Available value is 'affinity', 'anti-affinity'",
-				Required:    true,
-				ForceNew:    true,
+				Type: schema.TypeString,
+				Description: `
+Server group policy. Available value is 'affinity' 'anti-affinity' 'soft-anti-affinity'.
```
Comment (IsabellaMerkulova) [inline gcore/resource_gcore_servergroup.go:93]: The same comments on description as previous

File: gcore/resource_gcore_servergroup.go [hunk 8]
```diff
@@ -2,21 +2,29 @@ package gcore
...
+	"github.com/G-Core/gcorelabscloud-go/gcore/task/v1/tasks"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
 const (
-	serverGroupsPoint = "servergroups"
+	serverGroupsPoint          = "servergroups"
+	putToServerGroupTimeout    = 1200
+	removeToServerGroupTimeout = 1200
```
Comment (IsabellaMerkulova) [inline gcore/resource_gcore_servergroup.go:20]: removeToServerGroupTimeout -> removeFromServerGroupTimeout

File: docs/resources/servergroup.md (hunk 2 above)
Comment (alexk53) [reply docs/resources/servergroup.md:98]: It should be unique only inside one instance

File: docs/resources/servergroup.md [hunk 9]
```diff
@@ -12,29 +11,139 @@ Represent server group resource
...
+
+**Soft Anti-affinity rule**:
+
+Attempts to restrict instances belonging to the server group to the same host. Where it is not possible
+to schedule all instances on one host, they will be scheduled together on as few hosts as possible.
+
+**Anti-affinity**: 
+
+Boost performance by grouping related instances together. They will be hosted on the same physical server for
+faster interaction, enhancing communication and resource sharing.
```
Comment (IsabellaMerkulova) [inline docs/resources/servergroup.md:142]: Actually I didn't like soft anti affinity description, beacuse of "they will be scheduled together on as few hosts as possible."

so I decided to review using chat gpt, here is the result:

Affinity Rule:

Strictly isolates instances across different physical servers, minimizing simultaneous failures for critical applications. This rule is recommended for enhanced fault tolerance.

Soft Anti-affinity Rule:

Attempts to place instances belonging to the same server group on a single host. If this is not possible, instances will be scheduled on the minimum number of hosts required.

Anti-affinity Rule:

Improves performance by grouping related instances on the same physical server to enhance communication, resource sharing, and faster interaction.

**Pull request merged**